### PR TITLE
BUGFIX: Fix projection database schema

### PR DESCRIPTION
**Pull request by bwaidelich**
This is a follow-up to #4730 fixing the database schema for binary columns.

Without this fix, calling `ProjectionInterface::setUp()` will drop and re-add a lot of columns!

Furthermore, this fixes the `origindimensionspacepointhash` column of the `AssetUsageRepository` to correctly use the `DbalSchemaFactory::columnForDimensionSpacePointHash`. Previously it used `TEXT` without a length wich lead to the index from being recreated every setup.

### CODE REVIEW

**Comment by bwaidelich** [comment]
This is a draft because it depends on #4845

**Review by dlubitz — APPROVED** [review]
LGTM. Thank you!

**Review by kitsunet — APPROVED** [review]

File: Neos.Neos/Classes/AssetUsage/Projection/AssetUsageRepository.php [hunk 1]
```diff
@@ -73,8 +73,7 @@ private function databaseSchema(): Schema
             ->addIndex(['originalassetid'])
             ->addIndex(['contentstreamid'])
             ->addIndex(['nodeaggregateid'])
-            ->addIndex(['origindimensionspacepointhash']);
-        ;
+            ->addIndex(['origindimensionspacepointhash'], options: ['lengths' => [768]]);
```
Comment (kitsunet) [inline Neos.Neos/Classes/AssetUsage/Projection/AssetUsageRepository.php]: wait why do we need that much length here?

Comment (bwaidelich) [reply Neos.Neos/Classes/AssetUsage/Projection/AssetUsageRepository.php]: It's just what the db applied if it was not specified.
I was surprised that the hash column was changed to `TEXT` that is basically unbound.. Maybe we can limit the max length in our PHP VOs already to a sensible value. Then we could reduce column and index sizes, too

Comment (kitsunet) [reply Neos.Neos/Classes/AssetUsage/Projection/AssetUsageRepository.php]: Good that we talked about it then :) Apparently an oversight on my side, see 
https://github.com/neos/neos-development-collection/pull/4847/files/73e4d80eae3d584422c09c0ab5837e81050faf9f#diff-9139cc843ef200a3a9ed44f6f009526aecf42bb03934f02722e2d93e0fb2cc47R66

both ODSP and the hash column use the `columnForDimensionSpacePoint` spec, which is clearly wrong for the hash, sorry!

Comment (kitsunet) [reply Neos.Neos/Classes/AssetUsage/Projection/AssetUsageRepository.php]: Added a commit to fix it

Comment (mhsdesign) [reply Neos.Neos/Classes/AssetUsage/Projection/AssetUsageRepository.php]: So with your fix @kitsunet https://github.com/neos/neos-development-collection/pull/4847/commits/dc6c9e1a4c3dcbb2179925844432049eeddab007 should bastis be reverted?

Also ci fails now:

> PDOException: SQLSTATE[HY000]: General error: 1089 Incorrect prefix key; the used key part isn't a string, the used length is longer than the key part, or the storage engine doesn't support unique prefix keys in /home/runner/work/neos-development-collection/neos-development-collection/neos-development-distribution/Packages/Libraries/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php:55

Comment (kitsunet) [reply Neos.Neos/Classes/AssetUsage/Projection/AssetUsageRepository.php]: right, yeah, it must

**Comment by kitsunet** [comment]
Much betterer I think

**Comment by mhsdesign** [comment]
Adjusted the description above and will merge this ;)

**Pull request merged**